profiling/src/internal/observation/observations.rs

```diff
@@ -7,43 +7,58 @@ use super::super::Sample;
 use super::trimmed_observation::{ObservationLength, TrimmedObservation};
 use crate::internal::Timestamp;
 use std::collections::HashMap;
+use crate::collections::identifiable::*;
+use lz4_flex::frame::FrameEncoder;
+use std::io::Write;
 
 struct NonEmptyObservations {
     aggregated_data: HashMap<Sample, TrimmedObservation>,
```

Comment:
Would it be worth doing something for the aggregated data also? It would not be hard to make a poor man's bump allocator for this particular type.

Reply:
I think it's worth a try as part of the "reduce/eliminate all allocations", although I'd probably suggest doing it as a separate PR because it's a quite different solution: Unlike the timeline data, we do need to access and mutate these observations.

Reply:
Indeed maybe it's easier than what I was thinking. We could also pre-size both the map as well as the vec to make sure they started at a reasonable size instead of having to do a bunch of resizes with the first few hundred samples.